### PR TITLE
OCM-7172 | ci: Automation for hcp nodepool security groups support

### PR DESCRIPTION
**Pull request by xueli181114**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story (OCM-xxxx)*:
Fixes # [OCM-7172](https://issues.redhat.com//browse/OCM-7172)

**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Build
- [x] CI
- [ ] Documentation
- [ ] Performance
- [ ] Refactor
- [ ] Style
- [ ] Unit tests
- [ ] Subsystem tests

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

File: tests/tf-manifests/rhcs/machine-pools-hcp/main.tf
[the remark this answers is not in the document]
Comment (xueli181114) [reply tests/tf-manifests/rhcs/machine-pools-hcp/main.tf:1]: @radtriste Yes. Let me update it to `machine-pools/[rosa-classic|rosa-hcp]`

**Comment by xueli181114** [comment]
```
lixue@Xue-Lis-MacBook-Pro terraform-provider-rhcs % ginkgo -focus (@OCP-73068|@OCP-73069) tests/e2e
zsh: no matches found: (@OCP-73068|@OCP-73069)
lixue@Xue-Lis-MacBook-Pro terraform-provider-rhcs % ginkgo -focus '(@OCP-73068|@OCP-73069)' tests/e2e
time="2024-04-03T19:10:14+08:00" level=info msg="Running against env staging with gateway url https://api.stage.openshift.com"
time="2024-04-03T19:10:14+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-ad : map[admin_enabled:false autoscale:true byok:true byovpc:true channel_group:candidate cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.2xlarge etcd_encryption:true fips:false kms_key_arn:true labeling:false major_version:4.15 multi_az:true oidc_config:managed private:false product_id:rosa proxy:true region:us-west-2 sts:true tagging:true unified_acc_role_path: version: version_pattern:latest zones:a,b,c]"
Running Suite: e2e tests suite - /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/e2e
================================================================================================
Random Seed: 1712142608

Will run 2 of 63 specs
time="2024-04-03T19:10:14+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-ad : map[admin_enabled:false autoscale:true byok:true byovpc:true channel_group:candidate cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.2xlarge etcd_encryption:true fips:false kms_key_arn:true labeling:false major_version:4.15 multi_az:true oidc_config:managed private:false product_id:rosa proxy:true region:us-west-2 sts:true tagging:true unified_acc_role_path: version: version_pattern:latest zones:a,b,c]"
time="2024-04-03T19:10:14+08:00" level=info msg="Running terraform init against the dir /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/clusters/rosa-hcp"
time="2024-04-03T19:10:18+08:00" level=info msg="Running terraform output against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/clusters/rosa-hcp"
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
P [PENDING]
TF Test Verfication/Post day 1 tests Author:xueli-Critical-OCP-69145 @OCP-69145 @xueli Apply to change security group will be forbidden [NonHCPCluster]
/Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/e2e/verfication_post_day1_test.go:261
------------------------------
SSSSSSSSSSSSSSSStime="2024-04-03T19:10:20+08:00" level=info msg="Running terraform init against the dir /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp"
time="2024-04-03T19:10:20+08:00" level=info msg="Running terraform init against the dir /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/aws/vpc"
time="2024-04-03T19:10:24+08:00" level=info msg="Running terraform output against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/aws/vpc"
time="2024-04-03T19:10:26+08:00" level=info msg="Running terraform init against the dir /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/aws/security-groups"
time="2024-04-03T19:10:29+08:00" level=info msg="Running terraform output against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/aws/security-groups"
time="2024-04-03T19:10:31+08:00" level=info msg="Running terraform output against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/aws/security-groups"
time="2024-04-03T19:10:33+08:00" level=info msg="Running terraform apply against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp with args [-var additional_security_groups=[\"sg-0d47d112193d412fb\",\"sg-0ca1253a4325c08c5\",\"sg-08016e54f2f0e7590\",\"sg-0174bf5c097e39e56\"] -var auto_repair=false -var name=ocp-73068-2 -var url=https://api.stage.openshift.com -var machine_type=r5.xlarge -var autoscaling_enabled=false -var subnet_id=subnet-06bf16b5b55d68b12 -var cluster=2add20bo15cedljf52rppgq0f52jo3aj -var replicas=0]"
time="2024-04-03T19:10:39+08:00" level=info msg="Recording tfvars file /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp/terraform.tfvars"
time="2024-04-03T19:10:41+08:00" level=info msg="Running terraform apply against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp with args [-var url=https://api.stage.openshift.com -var additional_security_groups=[\"sg-0d47d112193d412fb\"]]"
time="2024-04-03T19:10:47+08:00" level=error msg="rhcs_hcp_machine_pool.mp: Refreshing state... [id=ocp-73068-2]\n\nTerraform used the selected providers to generate the following execution\nplan. Resource actions are indicated with the following symbols:\n  ~ update in-place\n\nTerraform will perform the following actions:\n\n  # rhcs_hcp_machine_pool.mp will be updated in-place\n  ~ resource \"rhcs_hcp_machine_pool\" \"mp\" {\n      ~ aws_node_pool     = {\n          ~ additional_security_group_ids = [\n                \"sg-0d47d112193d412fb\",\n              - \"sg-0ca1253a4325c08c5\",\n              - \"sg-08016e54f2f0e7590\",\n              - \"sg-0174bf5c097e39e56\",\n            ]\n            # (2 unchanged attributes hidden)\n        }\n        id                = \"ocp-73068-2\"\n        name              = \"ocp-73068-2\"\n      ~ status            = {\n          ~ current_replicas = 0 -> (known after apply)\n          ~ message          = \"WaitingForAvailableMachines\" -> (known after apply)\n        } -> (known after apply)\n        # (7 unchanged attributes hidden)\n    }\n\nPlan: 0 to add, 1 to change, 0 to destroy.\nrhcs_hcp_machine_pool.mp: Modifying... [id=ocp-73068-2]\n\nError: Provider produced inconsistent result after apply\n\nWhen applying changes to rhcs_hcp_machine_pool.mp, provider\n\"provider[\\\"terraform.local/local/rhcs\\\"]\" produced an unexpected new value:\n.aws_node_pool.additional_security_group_ids: new element 1 has appeared.\n\nThis is a bug in the provider, which should be reported in the provider's own\nissue tracker.\n\nError: Provider produced inconsistent result after apply\n\nWhen applying changes to rhcs_hcp_machine_pool.mp, provider\n\"provider[\\\"terraform.local/local/rhcs\\\"]\" produced an unexpected new value:\n.aws_node_pool.additional_security_group_ids: new element 2 has appeared.\n\nThis is a bug in the provider, which should be reported in the provider's own\nissue tracker.\n\nError: Provider produced inconsistent result after apply\n\nWhen applying changes to rhcs_hcp_machine_pool.mp, provider\n\"provider[\\\"terraform.local/local/rhcs\\\"]\" produced an unexpected new value:\n.aws_node_pool.additional_security_group_ids: new element 3 has appeared.\n\nThis is a bug in the provider, which should be reported in the provider's own\nissue tracker."
time="2024-04-03T19:10:47+08:00" level=info msg="Running terraform destroy against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp"
args:  /usr/local/bin/terraform destroy -auto-approve -no-color -var url=https://api.stage.openshift.com -var additional_security_groups=["sg-0d47d112193d412fb"]
time="2024-04-03T19:10:53+08:00" level=info msg="Deleting tfvars file /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp/terraform.tfvars"
time="2024-04-03T19:10:53+08:00" level=info msg="Running terraform destroy against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp"
args:  /usr/local/bin/terraform destroy -auto-approve -no-color -var url=https://api.stage.openshift.com -var additional_security_groups=["sg-0d47d112193d412fb"]

Error: No value for required variable

  on variable.tf line 1:
   1: variable "cluster" {

The root module input variable "cluster" is not set, and has no default
value. Use a -var or -var-file command line argument to provide a value for
this variable.

Error: No value for required variable

  on variable.tf line 8:
   8: variable "name" {

The root module input variable "name" is not set, and has no default value.
Use a -var or -var-file command line argument to provide a value for this
variable.
time="2024-04-03T19:10:53+08:00" level=error msg="exit status 1: var.cluster\n  Enter a value: var.name\n  Enter a value: "

------------------------------
• [FAILED] [33.569 seconds]
TF Test Create HCP Machine Pool test [It] Author:xueli-High-OCP-73068 @OCP-73068 @xueli Create nodepool with security groups via RHCS will work well [NonClassicCluster, day2, High]
/Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/e2e/hcp_machine_pool_test.go:38

  Timeline >>
  STEP: Get vpc output @ 04/03/24 19:10:20.603
  STEP: Prepare additional security groups @ 04/03/24 19:10:26.011
  STEP: Verify the parameters of the created machinepool @ 04/03/24 19:10:39.689
  STEP: Update security groups is not allowed to a machinepool @ 04/03/24 19:10:41.953
  [FAILED] in [It] - /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/e2e/hcp_machine_pool_test.go:100 @ 04/03/24 19:10:53.643
  << Timeline

  [FAILED] Expected
      <string>: rhcs_hcp_machine_pool.mp: Refreshing state... [id=ocp-73068-2]
      
      Terraform used the selected providers to generate the following execution
      plan. Resource actions are indicated with the following symbols:
        ~ update in-place
      
      Terraform will perform the following actions:
      
        # rhcs_hcp_machine_pool.mp will be updated in-place
        ~ resource "rhcs_hcp_machine_pool" "mp" {
            ~ aws_node_pool     = {
                ~ additional_security_group_ids = [
                      "sg-0d47d112193d412fb",
                    - "sg-0ca1253a4325c08c5",
                    - "sg-08016e54f2f0e7590",
                    - "sg-0174bf5c097e39e56",
                  ]
                  # (2 unchanged attributes hidden)
              }
              id                = "ocp-73068-2"
              name              = "ocp-73068-2"
            ~ status            = {
                ~ current_replicas = 0 -> (known after apply)
                ~ message          = "WaitingForAvailableMachines" -> (known after apply)
              } -> (known after apply)
              # (7 unchanged attributes hidden)
          }
      
      Plan: 0 to add, 1 to change, 0 to destroy.
      rhcs_hcp_machine_pool.mp: Modifying... [id=ocp-73068-2]
      
      Error: Provider produced inconsistent result after apply
      
      When applying changes to rhcs_hcp_machine_pool.mp, provider
      "provider[\"terraform.local/local/rhcs\"]" produced an unexpected new value:
      .aws_node_pool.additional_security_group_ids: new element 1 has appeared.
      
      This is a bug in the provider, which should be reported in the provider's own
      issue tracker.
      
      Error: Provider produced inconsistent result after apply
      
      When applying changes to rhcs_hcp_machine_pool.mp, provider
      "provider[\"terraform.local/local/rhcs\"]" produced an unexpected new value:
      .aws_node_pool.additional_security_group_ids: new element 2 has appeared.
      
      This is a bug in the provider, which should be reported in the provider's own
      issue tracker.
      
      Error: Provider produced inconsistent result after apply
      
      When applying changes to rhcs_hcp_machine_pool.mp, provider
      "provider[\"terraform.local/local/rhcs\"]" produced an unexpected new value:
      .aws_node_pool.additional_security_group_ids: new element 3 has appeared.
      
      This is a bug in the provider, which should be reported in the provider's own
      issue tracker.
  to contain substring
      <string>: Attribute aws_additional_security_group_ids, cannot be changed
  In [It] at: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/e2e/hcp_machine_pool_test.go:100 @ 04/03/24 19:10:53.643
------------------------------
time="2024-04-03T19:10:53+08:00" level=info msg="Running terraform init against the dir /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp"
time="2024-04-03T19:10:54+08:00" level=info msg="Running terraform init against the dir /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/aws/vpc"
time="2024-04-03T19:10:57+08:00" level=info msg="Running terraform output against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/aws/vpc"
time="2024-04-03T19:10:59+08:00" level=info msg="Running terraform apply against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp with args [-var replicas=0 -var autoscaling_enabled=false -var subnet_id=subnet-06bf16b5b55d68b12 -var cluster=2add20bo15cedljf52rppgq0f52jo3aj -var name=ocp-73069 -var url=https://api.stage.openshift.com -var machine_type=r5.xlarge -var additional_security_groups=[\"sg-fake\"] -var auto_repair=false]"
time="2024-04-03T19:11:04+08:00" level=error msg="Terraform used the selected providers to generate the following execution\nplan. Resource actions are indicated with the following symbols:\n  + create\n\nTerraform will perform the following actions:\n\n  # rhcs_hcp_machine_pool.mp will be created\n  + resource \"rhcs_hcp_machine_pool\" \"mp\" {\n      + auto_repair       = false\n      + autoscaling       = {\n          + enabled = false\n        }\n      + availability_zone = (known after apply)\n      + aws_node_pool     = {\n          + additional_security_group_ids = [\n              + \"sg-fake\",\n            ]\n          + instance_profile              = (known after apply)\n          + instance_type                 = \"r5.xlarge\"\n        }\n      + cluster           = \"2add20bo15cedljf52rppgq0f52jo3aj\"\n      + current_version   = (known after apply)\n      + id                = (known after apply)\n      + name              = \"ocp-73069\"\n      + replicas          = 0\n      + status            = (known after apply)\n      + subnet_id         = \"subnet-06bf16b5b55d68b12\"\n    }\n\nPlan: 1 to add, 0 to change, 0 to destroy.\n\nChanges to Outputs:\n  + autoscaling_enabled = {\n      + enabled      = false\n      + max_replicas = null\n      + min_replicas = null\n    }\n  + cluster_id          = \"2add20bo15cedljf52rppgq0f52jo3aj\"\n  + machine_pool_id     = (known after apply)\n  + machine_type        = \"r5.xlarge\"\n  + name                = \"ocp-73069\"\n  + replicas            = 0\nrhcs_hcp_machine_pool.mp: Creating...\n\nError: Cannot create machine pool\n\n  with rhcs_hcp_machine_pool.mp,\n  on main.tf line 26, in resource \"rhcs_hcp_machine_pool\" \"mp\":\n  26: resource \"rhcs_hcp_machine_pool\" \"mp\" {\n\nCannot create machine pool for cluster '2add20bo15cedljf52rppgq0f52jo3aj':\nstatus is 400, identifier is '400', code is 'CLUSTERS-MGMT-400' and operation\nidentifier is '5812c10e-989a-4a31-9f58-69a3d604dc0b': Provided Additional\nSecurity Group 'sg-fake' is not attached to VPC 'vpc-05669ff2493e1e747'"
time="2024-04-03T19:11:04+08:00" level=info msg="Running terraform plan against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp with args [-var replicas=0 -var autoscaling_enabled=false -var additional_security_groups=[\"sg-fake\",\"sg-fakeid0\",\"sg-fakeid1\",\"sg-fakeid2\",\"sg-fakeid3\",\"sg-fakeid4\",\"sg-fakeid5\",\"sg-fakeid6\",\"sg-fakeid7\",\"sg-fakeid8\",\"sg-fakeid9\",\"sg-fakeid10\"] -var auto_repair=false -var cluster=2add20bo15cedljf52rppgq0f52jo3aj -var name=ocp-73069 -var url=https://api.stage.openshift.com -var machine_type=r5.xlarge -var subnet_id=subnet-06bf16b5b55d68b12]"
time="2024-04-03T19:11:05+08:00" level=info msg="Running terraform destroy against the dir: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/machine-pools/hcp"
args:  /usr/local/bin/terraform destroy -auto-approve -no-color -var url=https://api.stage.openshift.com -var machine_type=r5.xlarge -var auto_repair=false -var cluster=2add20bo15cedljf52rppgq0f52jo3aj -var name=ocp-73069 -var replicas=0 -var autoscaling_enabled=false -var subnet_id=subnet-06bf16b5b55d68b12 -var additional_security_groups=["sg-fake","sg-fakeid0","sg-fakeid1","sg-fakeid2","sg-fakeid3","sg-fakeid4","sg-fakeid5","sg-fakeid6","sg-fakeid7","sg-fakeid8","sg-fakeid9","sg-fakeid10"]
• [FAILED] [11.971 seconds]
TF Test Create HCP Machine Pool test [It] Author:xueli-High-OCP-73069 @OCP-73069 @xueli Create nodepool with security groups via RHCS will work well [NonClassicCluster, day2, Low]
/Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/e2e/hcp_machine_pool_test.go:125

  Timeline >>
  STEP: Get vpc output @ 04/03/24 19:10:54.176
  STEP: Prepare additional security groups @ 04/03/24 19:10:59.523
  STEP: Run terraform apply cannot work with invalid sg IDs @ 04/03/24 19:10:59.523
  STEP: Terraform plan with too many sg IDs cannot work @ 04/03/24 19:11:04.899
  [FAILED] in [It] - /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/e2e/hcp_machine_pool_test.go:166 @ 04/03/24 19:11:05.351
  << Timeline

  [FAILED] Expected an error to have occurred.  Got:
      <nil>: nil
  In [It] at: /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/e2e/hcp_machine_pool_test.go:166 @ 04/03/24 19:11:05.351
------------------------------
SSSSSSS

Summarizing 2 Failures:
  [FAIL] TF Test Create HCP Machine Pool test [It] Author:xueli-High-OCP-73068 @OCP-73068 @xueli Create nodepool with security groups via RHCS will work well [NonClassicCluster, day2, High]
  /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/e2e/hcp_machine_pool_test.go:100
  [FAIL] TF Test Create HCP Machine Pool test [It] Author:xueli-High-OCP-73069 @OCP-73069 @xueli Create nodepool with security groups via RHCS will work well [NonClassicCluster, day2, Low]
  /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/e2e/hcp_machine_pool_test.go:166

Ran 2 of 63 Specs in 51.622 seconds
FAIL! -- 0 Passed | 2 Failed | 1 Pending | 60 Skipped
--- FAIL: TestRHCSProvider (51.63s)
You're using deprecated Ginkgo functionality:
=============================================
  --ginkgo.slow-spec-threshold is deprecated --slow-spec-threshold has been deprecated and will be removed in a future version of Ginkgo.  This feature has proved to be more noisy than useful.  You can use --poll-progress-after, instead, to get more actionable feedback about potentially slow specs and understand where they might be getting stuck.

To silence deprecations that can be silenced set the following environment variable:
  ACK_GINKGO_DEPRECATIONS=2.13.2

FAIL

Ginkgo ran 1 suite in 57.021419156s

Test Suite Failed
```
Testing failed due to known issues.

File: tests/tf-manifests/rhcs/machine-pools-hcp/main.tf
[the remark this answers is not in the document]
Comment (xueli181114) [reply tests/tf-manifests/rhcs/machine-pools-hcp/main.tf:1]: Updated

**Comment by xueli181114** [comment]
@yasun1 @oriAdler @radtriste Please help to review

**Review by radtriste — APPROVED** [review]

**Pull request merged**